x-pack/plugins/aiops/public/components/log_categorization/use_discover_links.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import rison from 'rison-node';
+import moment from 'moment';
+
+import type { TimeRangeBounds } from '@kbn/data-plugin/common';
+import { useAiopsAppContext } from '../../hooks/use_aiops_app_context';
+import type { Category } from './use_categorize_request';
+import type { QueryMode } from './category_table';
+import type { AiOpsIndexBasedAppState } from '../explain_log_rate_spikes/explain_log_rate_spikes_app_state';
+
+export function useDiscoverLinks() {
+  const {
+    http: { basePath },
+  } = useAiopsAppContext();
+
+  const openInDiscoverWithFilter = (
+    index: string,
+    field: string,
+    selection: Category[],
+    aiopsListState: Required<AiOpsIndexBasedAppState>,
+    timefilterActiveBounds: TimeRangeBounds,
+    mode: QueryMode,
+    category?: Category
+  ) => {
+    const selectedRows = category === undefined ? selection : [category];
+
+    const _g = rison.encode({
+      time: {
+        from: moment(timefilterActiveBounds.min?.valueOf()).toISOString(),
+        to: moment(timefilterActiveBounds.max?.valueOf()).toISOString(),
+      },
+    });
+
+    const _a = rison.encode({
+      filters: [
+        ...aiopsListState.filters,
+        {
+          query: {
+            bool: {
+              [mode]: selectedRows.map(({ key: query }) => ({
+                match: {
+                  [field]: {
+                    auto_generate_synonyms_phrase_query: false,
+                    fuzziness: 0,
+                    operator: 'and',
+                    query,
+                  },
+                },
+              })),
+            },
+          },
+        },
+      ],
+      index,
+      interval: 'auto',
+      query: {
+        language: aiopsListState.searchQueryLanguage,
+        query: aiopsListState.searchString,
+      },
+    });
+
+    let path = basePath.get();
+    path += '/app/discover#/';
+    path += '?_g=' + _g;
+    path += '&_a=' + encodeURIComponent(_a);
+    window.open(path, '_blank');
```

Comment:
Is it a requirement to open a new tab? I reckon it should be up to the user

Reply:
Personally I think opening in a new tab here is better because of the lack of URL state, which means if the user clicks to navigate to Discover and then wants to go back, they'll have to re-select everything on the page and run the analysis again which could take a long time.
But I understand the argument for making these normal hrefs so the user can choose whether to open in a new tab.
@peteharverson @walterra what are your opinions on this?

Reply:
We have a similar problem in Explain Log Rate Spikes. There we don't open a new window when drilling down to Discover. If you go back you lost all settings and need to rerun the analysis. On top of that, not sure about the implementation of the table actions, but opening in a new tab there doesn't work even if you want to. So I'm more leaning towards opening a new window/tab by default until we sort out that the back button works and gets you back to the exact view you were at before.

Reply:
Agree with @darnautov that ultimately it will be better to make these regular links where the user can decide whether or not to open in the same tab. But until the page state is persisted in the URL so that Back button behavior works, best to open in a new tab.